test/Mono.Linker.Tests.Cases/Attributes/TypeWithDynamicInterfaceCastableImplementationAttributeIsKept.cs

```diff
@@ -23,7 +21,7 @@ public class TypeWithDynamicInterfaceCastableImplementationAttributeIsKept
 	{
 		public static void Main ()
 		{
-#if NETCOREAPP
+#if SUPPORTS_DEFAULT_INTERFACE_METHODS
```

Comment:
It looks like the `#if NETCOREAPP` was added because this code wouldn't compile against .NET Framework.  There are types such as `IDynamicInterfaceCastable` that don't exist.

We haven't pulled this test into our fork yet.  And it looks like removing all of the #ifs in this file wouldn't cause us any issues if we did have this test.

Is it OK if we leave these #ifs in the PR and you can remove them later if you don't want them?  I don't care either way.  I mostly want to avoid having this PR held up longer.

Reply:
I don't think this has much to do with DIM as it's simply API available since net5. Would changing the define to NET_5_0 work for you?

Reply:
I was thinking about this more last night and you're right, this is really specific to net5, not DIM.  I'll fix it.

Reply:
I changed it back to `NETCOREAPP` instead of `NET_5_0`.